src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todo: Todo;
+  loadingTodos: number[];
+  handleRemoveTodo: (id: number) => void;
+  handleTodoStatusUpdate: (todo: Todo) => void;
+  handleTodoTitleUpdate: (todo: Todo, title: string) => Promise<void>;
+}
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  loadingTodos,
+  handleRemoveTodo,
+  handleTodoStatusUpdate,
+  handleTodoTitleUpdate,
+}) => {
+  const { id, title, completed } = todo;
+
+  const [newTitle, setNewTitle] = useState(title);
+  const [isDoubleClicked, setIsDoubleClicked] = useState(false);
+
+  const todoInputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (todoInputRef.current) {
+      todoInputRef.current.focus();
+    }
+  }, [isDoubleClicked]);
+
+  const handleDoubleClick = () => {
+    setIsDoubleClicked(true);
+  };
+
+  const handleTodoInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTitle(event.target.value);
+  };
+
+  const saveChanges = () => {
+    handleTodoTitleUpdate(todo, newTitle);
+    setIsDoubleClicked(false);
+  };
+
+  const handleOnBlur = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!newTitle.trim()) {
+      handleRemoveTodo(id);
+    }
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!newTitle) {
+      handleRemoveTodo(id);
+    }
+
+    if (newTitle === title) {
+      setNewTitle(title);
+      setIsDoubleClicked(false);
+
+      return;
+    }
+
+    saveChanges();
+  };
+
+  const handleCancel = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Escape') {
+      setNewTitle(title);
+      setIsDoubleClicked(false);
+    }
```

Comment:
```suggestion
    if (event.key !== 'Escape') {
      return;
    }
    
    setNewTitle(title);
    setIsDoubleClicked(false);
```

in case you'll have more logic here, it would be a better approach since you'll have less inner scopes